### ISSUE TITLE
P1 audit-v2: remove per-boundary ceiling bias from partial-supply condition loss

### ISSUE THREAD

**Issue by Datengu** [Datengu/anthrosim#326]
## Scientific audit v2 finding

The v9 elapsed-time repair for M3 condition response correctly partitions the configured reference-quarter **maximum condition-loss budget** across arbitrary `resources.periodsPerYear`, but the actual under-supply loss applies an integer ceiling separately at every M3 boundary:

```rust
let deficit = 1000 - supplied_permille;
let loss_numerator = deficit * interval_max_loss;
let loss = if loss_numerator == 0 {
    0
} else {
    loss_numerator.div_ceil(1000)
};
condition = condition.saturating_sub(loss);
```

This reintroduces a strong resource-clock dependence whenever supply is partial but close to complete.

### Evidence

Audited executable tree: model semantics `anthrosim-model-semantics-v19`; protected main at discovery `fdeb66ed0e05683fd5092f3e1ec8407df1bbcfe4` (the previous executable code is unchanged from `06f1a059...`).

`reference_quarter_quantity_for_interval(...)` is internally coherent: for a configured reference-quarter maximum loss of 100, every allowed partition sums to exactly 400 maximum-loss points over the complete year.

The problem occurs in the next step, where each interval independently rounds any positive fractional loss upward to at least one whole condition point.

For a constant supply deficit of just **1 permille** (`supplied_permille = 999`) and reference-quarter max loss 100:

| M3 settlements/year | Sum of interval max-loss budgets | Realized yearly condition loss from current ceiling rule |
|---:|---:|---:|
| 1 | 400 | 1 |
| 4 | 400 | 4 |
| 12 | 400 | 12 |
| 52 | 400 | 52 |
| 365 | 400 | 365 |

Across every allowed `periodsPerYear = 1..365`, the same constant 999/1000 supply fraction therefore produces **365 distinct annual condition losses**, monotonically ranging from 1 to 365 solely because of settlement cadence.

The underlying unrounded annual loss represented by the configured linear rule is only `1/1000 * 400 = 0.4` condition points. The model cannot represent a fractional point in authoritative state, but repeated per-boundary ceiling converts one sub-unit annual effect into up to 365 whole points.

The bias is not limited to a one-permille edge case. Exact independent arithmetic using the current source equations gives:

| Supply deficit (permille) | P=1 | P=4 | P=12 | P=52 | P=365 |
|---:|---:|---:|---:|---:|---:|
| 10 | 4 | 4 | 12 | 52 | 365 |
| 50 | 20 | 20 | 24 | 52 | 365 |
| 100 | 40 | 40 | 48 | 52 | 365 |
| 500 | 200 | 200 | 204 | 209 | 365 |
| 999 | 400 | 400 | 400 | 400 | 400 |

So finer M3 settlement can make mild or moderate shortages dramatically more physiologically damaging even though elapsed exposure, annual need, supply fraction, and the configured annual/reference-quarter response budget are held fixed.

### Why existing acceptance did not catch it

#204 / PR #241 explicitly closed the earlier temporal-resolution defect with the acceptance statement that condition recovery/loss uses elapsed M3 interval allocation and that controlled 1/4/12/365 partitions preserve the intended condition response.

The current regression `condition_response_does_not_multiply_with_resource_partition` checks:

- **fully supplied** recovery; and
- **zero supplied** loss.

Both endpoint cases hide this defect:

- full supply uses the exactly partitioned recovery quantity directly;
- zero supply has deficit 1000, so `ceil(1000 * interval_max_loss / 1000) == interval_max_loss` and the interval sums reconcile exactly.

No current test exercises a fixed **partial** supply fraction across temporal partitions.

### Scientific consequence

`resources.periodsPerYear` is documented after #204 as the M3 resource integration/settlement clock rather than a multiplier for physiological response. Under the present arithmetic, however, increasing only that clock can:

- multiply annual condition deterioration under mild scarcity;
- change condition-mediated mortality through the resulting condition trajectory;
- change M4 condition pressure and migration behavior;
- alter cohort survival and demographic/resource feedback;
- make temporal-resolution sensitivity appear to show stronger scarcity effects that are actually integer-rounding accumulation.

This is a model-semantics defect, not merely small numerical noise: the 1-permille example changes annual loss from 1 to 365 condition points across the allowed clock range.

Severity: **P1** because it violates the explicit temporal-response contract established by #204 and can materially change survival/migration under otherwise equivalent partial-supply histories.

### Required work

- Replace independent per-boundary upward rounding with a deterministic elapsed-time accumulation rule that preserves the intended complete-interval/annual partial-supply response.
- Possible approaches include a carried fractional condition-loss accumulator, a cumulative fixed-point response calculation, or another exact integer scheme whose result is invariant (or convergent under one explicitly declared rounding contract) to arbitrary M3 partitioning when the underlying supply trajectory is held constant.
- Do not fix this by simply flooring every interval; repeated flooring would create the opposite cadence-dependent bias.
- Define whether authoritative condition remains integer permille and, if so, where fractional response state is accumulated/rounded.
- Add metamorphic tests for fixed supply fractions across at least `periodsPerYear = 1, 4, 12, 52, 365`, including deficits 1, 10, 100, 500 and 1000 permille.
- Preserve exact behavior for the existing endpoint cases where intended.
- Re-test condition-mediated mortality and M4 condition-pressure outcomes under equivalent partial-supply histories.
- Re-run affected scientific references/controlled experiments and bump model-semantics provenance because authoritative trajectories will change.
- Update the M3 timing/response contract to state the exact fractional/integer accumulation rule.

### Acceptance

Holding the elapsed resource interval and household supply fraction trajectory fixed cannot multiply total condition deterioration merely by subdividing the same exposure into more M3 settlement boundaries. Any residual integer-state rounding has one explicit cumulative meaning rather than a fresh upward bias at every boundary.
